src/Altinn.App.Core/Implementation/AppResourcesSI.cs

```diff
@@ -104,66 +109,49 @@ public byte[] GetText(string org, string app, string textResource)
         /// <inheritdoc />
         public Application GetApplication()
         {
-            // Cache application metadata
-            if (_application != null)
-            {
-                return _application;
-            }
-
-            string filedata = string.Empty;
-            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ApplicationMetadataFileName;
             try
             {
-                if (File.Exists(filename))
+                var task = Task.Run(async () => await _appMetadata.GetApplicationMetadata());
+                task.Wait();
+                if (task.IsCompletedSuccessfully)
                 {
-                    filedata = File.ReadAllText(filename, Encoding.UTF8);
+                    return task.Result;
                 }
-
-                _application = JsonConvert.DeserializeObject<Application>(filedata)!;
-                return _application;
             }
             catch (Exception ex)
             {
-                _logger.LogError("Something went wrong when fetching application metadata. {0}", ex);
-                return null;
+                _logger.LogError(ex, "Something went wrong fetching application metadata");
+                return null;   
             }
+            _logger.LogError("Something went wrong fetching application metadata");
+            return null;
```

Comment:
@RonnyB71 should we just ignore these exceptions here as well? Would change how the class works but we are no longer returning null on a non-nullable return type

Reply:
Yes, I think so. Does not make sense to swallow the exception here and let it blow up somewhere else.

Reply:
And let's create version 8 just to signal that we are changing behavior even though it would crash in a different place anyway.